sunbeam-python/sunbeam/plugins/observability/plugin.py

```diff
@@ -298,6 +298,76 @@ def run(self, status: Optional[Status] = None) -> Result:
         return Result(ResultType.COMPLETED)
 
 
+class DeployGrafanaAgentK8sStep(BaseStep, JujuStepHelper):
+    """Deploy Grafana Agent k8s using Terraform"""
+
+    def __init__(
+        self,
+        plugin: "ObservabilityPlugin",
+        tfhelper: TerraformHelper,
+        tfhelper_cos: TerraformHelper,
+        jhelper: JujuHelper,
+    ):
+        super().__init__("Deploy Grafana Agent k8s", "Deploying Grafana Agent k8s")
+        self.tfhelper = tfhelper
+        self.tfhelper_cos = tfhelper_cos
+        self.jhelper = jhelper
+        self.model = OPENSTACK_MODEL
+        self.read_config = lambda: plugin.get_plugin_info().get(
+            "grafana-agent-k8s-config", {}
+        )
+        self.update_config = lambda c: plugin.update_plugin_info(
+            {"grafana-agent-k8s-config": c}
+        )
+
+    def run(self, status: Optional[Status] = None) -> Result:
+        """Execute configuration using terraform."""
+        app = "grafana-agent-k8s"
+        cos_backend = self.tfhelper_cos.backend
+        cos_backend_config = self.tfhelper_cos.backend_config()
+        try:
+            config = self.read_config()
+        except ConfigItemNotFoundException as e:
+            LOG.exception("Failed deploying %s: unable to read config", app)
+            return Result(ResultType.FAILED, str(e))
+
+        tfvars = {
+            "cos-state-backend": cos_backend,
+            "cos-state-config": cos_backend_config,
+            "grafana-agent-k8s-channel": "latest/stable",
```

Comment:
Can you add a constant GRAFANA_AGENT_K8S_CHANNEL = "latest/stable" on the top and use that constant here.

Reply:
Updated